File: Models/ProductSerializableFactory.cs
using Hazelcast.Serialization;

namespace HazelcastCaching.Models
{
    public class ProductSerializableFactory : IDataSerializableFactory
    {
        public const int FactoryId = 1000;

        public IIdentifiedDataSerializable Create(int typeId)
        {
            if (typeId == Product.ClassId) return new Product();
            return null;
        }
    }
}

File: Java classes/ProductSerializableFactory.java
import com.hazelcast.nio.serialization.DataSerializableFactory;
import com.hazelcast.nio.serialization.IdentifiedDataSerializable;

public class ProductSerializableFactory implements DataSerializableFactory {
	public static final int FACTORY_ID = 1000;

    @Override
   public IdentifiedDataSerializable create(int typeId) {
       if (typeId == 100) {
           return new Product();
       }
       return null;
   }
}
